packages/core/data-transfer/lib/providers/local-strapi-destination-provider/index.ts

```diff
@@ -92,6 +100,54 @@ class LocalStrapiDestinationProvider implements IDestinationProvider {
     return mapSchemasValues(schemas);
   }
 
+  async getAssetsStream(): Promise<NodeJS.WritableStream> {
+    if (!this.strapi) {
+      throw new Error('Not able to stream Assets. Strapi instance not found');
+    }
+
+    const assetsDirectory = path.join(this.strapi.dirs.static.public, 'uploads');
+    const backupDirectory = path.join(
+      this.strapi.dirs.static.public,
+      `uploads_backup_${Date.now()}`
+    );
+
+    await fse.rename(assetsDirectory, backupDirectory);
+    await fse.mkdir(assetsDirectory);
+
+    return new Writable({
+      objectMode: true,
+      async final(next) {
+        await fse.rm(backupDirectory, { recursive: true, force: true });
+        next();
+      },
+      async write(chunk: IAsset, _encoding, callback) {
+        const entryPath = path.join(assetsDirectory, chunk.filename);
+        const writableStream = fse.createWriteStream(entryPath);
+
+        chunk.stream
+          .pipe(writableStream)
+          .on('close', callback)
+          .on('error', async (error: Error) => {
+            try {
+              await fse.rm(assetsDirectory, { recursive: true, force: true });
+              await fse.rename(backupDirectory, assetsDirectory);
+              this.destroy(
```

Comment:
I just wanted to highlight this so that you can give me your feedback. I've added the `destroy` method here because if there is an error we just want to stop the assets streaming (remember that the assets are transmitted one by one) and do a "rollback" deleting the newly created folder and restoring the previous one. However, if we don't stop the stream, an error will be thrown again and the assetsDirectory doesn't exist anymore so it will delete the backup assets directory (that had been renamed previously to do the rollback) and fail to rename the backupDirectory, that doesn't exist anymore (and in the end, losing both folders and all the backup assets). I hope that the explanation was clear and I would like to read your thoughts about this 😄

Reply:
That makes sense to me!